core/src/test/java/io/questdb/cairo/CreateTableTest.java

```diff
@@ -0,0 +1,191 @@
+    }
+
+    @Test(expected = SqlException.class)
+    public void testCreateTableAsSelectWithCastSymbolToStringAndIndexOnIt() throws Exception {
+        assertCompile("create table old(s symbol,l long, ts timestamp)");
+        assertQuery("s\tl\tts\n", "select * from new",
+                "create table new as (select * from old), index(s), cast(s as string)", null);
+    }
+
+    @Test(expected = SqlException.class)
+    public void testCreateTableAsSelectWithIndexOnSymbolCastedToString() throws Exception {
+        assertCompile("create table old(s symbol,l long, ts timestamp)");
+        assertQuery("s\tl\tts\n", "select * from new",
+                "create table new as (select * from old), cast(s as string), index(s)", null);
+    }
+
+    @Test
+    public void testCreateTableAsSelectWithMultipleInterleavedCastAndIndexes() throws Exception {
+        assertCompile("create table old(s string,sym symbol, ts timestamp)");
+        assertQuery("s\tsym\tts\n", "select * from new",
+                "create table new as (select * from old), cast(s as symbol), index(s), cast(ts as date), index(sym), cast(sym as symbol)", null);
+
+        assertColumnsIndexed("new", "s", "sym");
+    }
+
+    @Test
+    public void testCreateTableAsSelectWithMultipleInterleavedCastAndIndexesV2() throws Exception {
+        assertCompile("create table old(s string,sym symbol, ts timestamp)");
+        assertQuery("s\tsym\tts\n", "select * from new",
+                "create table new as (select * from old), cast(s as symbol), index(s), cast(ts as date), index(sym), cast(sym as symbol)", null);
+
+        assertColumnsIndexed("new", "s", "sym");
+    }
+
+    @Test
+    public void testCreateTableAsSelectWithMultipleInterleavedCastAndIndexesV3() throws Exception {
+        assertCompile("create table old(s string,sym symbol, ts timestamp)");
+        assertQuery("s\tsym\tts\n", "select * from new",
+                "create table new as (select * from old), index(s), cast(s as symbol), cast(ts as date), index(sym), cast(sym as symbol)", null);
+
+        assertColumnsIndexed("new", "s", "sym");
+    }
+
+    @Test
+    public void testCreateTableAsSelectInheritsColumnIndex() throws Exception {
+        assertCompile("create table old(s string,sym symbol index, ts timestamp)");
+        assertQuery("s\tsym\tts\n", "select * from new",
+                "create table new as (select * from old), index(s), cast(s as symbol), cast(ts as date)", null);
+
+        assertColumnsIndexed("new", "s");
+    }
+
+    private void assertColumnsIndexed(String tableName, String... columnNames) throws Exception {
+        assertMemoryLeak(() -> {
+            try (TableReader r = engine.getReader(AllowAllCairoSecurityContext.INSTANCE, tableName)) {
+                TableReaderMetadata metadata = r.getMetadata();
+                IntList indexed = new IntList();
+
+                for (int i = 0, cols = metadata.getColumnCount(); i < cols; i++) {
```

Comment:
minor: this for is equivalent to 1 line
```suggestion
                indexed.setPos(metadata.getColumnCount());
```

Reply:
Fixed